docs/source/user/getting_started/advanced_workflow_concepts.rst

```diff
@@ -0,0 +1,309 @@
+.. jinja:: first_ctx
+
+Advanced workflow concepts
+###########################
+
+Resources
+----------
+
+Requesting resources can be done using a ``resources`` block, either for the whole workflow at the top level,
+
+.. code-block:: yaml
+
+    resources:
+    any:
+      scheduler: sge # Setting the scheduler is not normally needed because a
+                     # `default_scheduler` will be set in the config file.
+      scheduler_args:
+        shebang_args: --login
+        options:
+          -l: short
+
+
+or at the task level
+
+.. code-block:: yaml
+
+  - schema: simulate_VE_loading_damask
+    resources:
+      any:
+      # This will use two cores for input file generators and output file parsers
+      num_cores: 2
+      main:
+      # Use 16 cores for the "main" part of the task (the simulation in this case)
+      num_cores: 16
+    inputs:
+        ...
+
+
+We can see from above that it is possible to request resources for subsets of the actions
+in a task schema. A full list of the different options you can select resources for is given below.
+
+- ``input_file_generator``
+- ``output_file_parser``
+- ``processing`` (a shortcut for ``input_file_generator`` +  ``output_file_parser``)
+- ``main`` (the main part of the action i.e. not ``processing``)
+- ``any`` (anything not already specified with any of the above options)
+
+These are used to choose resources (done at the workflow/task level),
+and also the same values can be used within the schema to select an ``environment``
+by ``scope`` e.g.
+
+.. code-block:: yaml
+
+    actions:
+    - environments:
+    - scope:
+        type: processing
+        environment: damask_parse_env
+    - scope:
+        type: main
+        environment: damask_env
+
+
+{{ app_name }} is then looking for a match within your environment definitions for the requested
+resources, and will run the command which matches those resources.
+
+There are lots of :ref:`resource options <reference/_autosummary/matflow.ResourceSpec:matflow.ResourceSpec>`
```

Comment:
We need to change the `matflow` to something that works for both in this URL; for the case of hpcflow it would be `hpcflow.app`, which corresponds to the `App.module` attribute, which in turn we expose as the `{{ app_module }}` jinja variable.